### PR TITLE
Verify Solana transactions

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
Now it just supports **SOL** transfer on Solana chain, for token transfer I will continue in another PR

related to #1171

### CODE REVIEW

File: src/services/chains/solana/transactionService.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import { normalizeAmount } from '../../../utils/utils';
+import {
+  i18n,
+  translationErrorMessagesKeys,
+} from '../../../utils/errorMessages';
+
+const SOLANA_NODE_RPC_URL = process.env.SOLANA_NODE_RPC_URL as string;
+const solana = new SolanaWeb3.Connection(SOLANA_NODE_RPC_URL);
+
+export const getSolanaTransactionDetailForNormalTransfer = async (params: {
```
Comment (aminlatifi) [inline src/services/chains/solana/transactionService.ts]:
```suggestion
export const getSolanaTransactionDetailForSolanaTransfer = async (params: {
```

File: src/services/chains/solana/transactionService.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  validateTransactionWithInputData,
+} from '../index';
+import { normalizeAmount } from '../../../utils/utils';
+import {
+  i18n,
+  translationErrorMessagesKeys,
+} from '../../../utils/errorMessages';
+
+const SOLANA_NODE_RPC_URL = process.env.SOLANA_NODE_RPC_URL as string;
+const solana = new SolanaWeb3.Connection(SOLANA_NODE_RPC_URL);
```
Comment (aminlatifi) [inline src/services/chains/solana/transactionService.ts]: Let's not put it here, this will be executed each time the module is imported. It means the whole app will crash if anything happens to it! 
Please move it inside a function. It's not a big deal if we instantiate the connection each time we do the validation.

File: src/services/chains/solana/transactionService.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  } catch (e) {
+    logger.error('getSolanaTransactionDetailForNormalTransfer error', e);
+    return null;
+  }
+};
+
+function getTransactionDetailForTokenTransfer(
+  input: TransactionDetailInput,
+): Promise<NetworkTransactionInfo> {
+  throw new Error();
```
Comment (aminlatifi) [inline src/services/chains/solana/transactionService.ts]:
```suggestion
  throw new Error('Not implemented!');
```

File: src/services/chains/solana/transactionService.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+}): Promise<NetworkTransactionInfo | null> => {
+  const TRANSFER_INSTRUCTION_TYPE = 'transfer';
+  const SOL_CURRENCY_TYPE = 'SOL';
+
+  try {
+    const result = await solana.getParsedTransaction(params.txHash);
+    const data = result?.transaction?.message?.instructions?.find(
+      instruction =>
+        (instruction as ParsedInstruction)?.parsed?.type ===
+        TRANSFER_INSTRUCTION_TYPE,
```
Comment (aminlatifi) [inline src/services/chains/solana/transactionService.ts]: Let's filter based on source and destination as well. I am not sure single transfer exists in a Solana transaction

File: src/services/chains/index.test.ts [hunk 5]
```diff
@@ -957,4 +871,47 @@ function getTransactionDetailTestCases() {
...
+        toAddress: 'DvWdrYYkwyM9mnTetpr3HBHUBKZ22QdbFEXQ8oquE7Zb',
+        timestamp: 1702931400 + ONE_DAY,
+        amount,
+      });
+    };
+    await assertThrowsAsync(
+      badFunc,
+      errorMessages.TRANSACTION_CANT_BE_OLDER_THAN_DONATION,
+    );
+  });
```
Comment (aminlatifi) [inline src/services/chains/index.test.ts:935]: Please add more tests if you can to cover other cases. You can use other transactions, like
3nzHwgxAu7mKw1dhGTVmqzY8Yet3kGWWqP5kr5D2fw1HzqPjqDGDe6xT5PguKXk8nAJcK4GpBEKWw7EzoLykKkCx

**Review by aminlatifi — CHANGES_REQUESTED** [review]

File: src/services/chains/solana/transactionService.ts (hunk 1 above)
Comment (mohammadranjbarz) [reply src/services/chains/solana/transactionService.ts]: Done

File: src/services/chains/solana/transactionService.ts (hunk 2 above)
Comment (mohammadranjbarz) [reply src/services/chains/solana/transactionService.ts]: Done

File: src/services/chains/solana/transactionService.ts (hunk 3 above)
Comment (mohammadranjbarz) [reply src/services/chains/solana/transactionService.ts]: Done

File: src/services/chains/solana/transactionService.ts (hunk 4 above)
Comment (mohammadranjbarz) [reply src/services/chains/solana/transactionService.ts]: Good idea, done

File: src/services/chains/index.test.ts (hunk 5 above)
Comment (mohammadranjbarz) [reply src/services/chains/index.test.ts:935]: @aminlatifi 
I just added another successful test case with your tx, for adding more test cases to cover all scenarios I need more time, does it needed for now or I can do that later?

**Review by aminlatifi — APPROVED** [review]

**Comment by aminlatifi** [comment]
Thanks @mohammadranjbarz

**Review by CarlosQ96 — APPROVED** [review]
Thanks Mohammad, all changes and feedback LGTM. 
We should test asap, maybe in staging to see if we don't have any edge case.
